test/Npgsql.Tests/ConnectionTests.cs

```diff
@@ -859,158 +829,143 @@ public void Notice()
-                RemoteCertificateValidationCallback callback2 = (sender, certificate, chain, errors) => true;
-                conn.UserCertificateValidationCallback = callback2;
-
-                conn.Open();
-                using (var conn2 = (NpgsqlConnection)((ICloneable)conn).Clone())
-                {
-                    Assert.That(conn2.ConnectionString, Is.EqualTo(conn.ConnectionString));
-                    Assert.That(conn2.ProvideClientCertificatesCallback, Is.SameAs(callback1));
-                    Assert.That(conn2.UserCertificateValidationCallback, Is.SameAs(callback2));
-                    conn2.Open();
-                }
-            }
+            };
+            using var conn = new NpgsqlConnection(csb.ConnectionString);
+            ProvideClientCertificatesCallback callback1 = certificates => { };
+            conn.ProvideClientCertificatesCallback = callback1;
+            RemoteCertificateValidationCallback callback2 = (sender, certificate, chain, errors) => true;
+            conn.UserCertificateValidationCallback = callback2;
+
+            conn.Open();
+            using var conn2 = (NpgsqlConnection)((ICloneable)conn).Clone();
+            Assert.That(conn2.ConnectionString, Is.EqualTo(conn.ConnectionString));
+            Assert.That(conn2.ProvideClientCertificatesCallback, Is.SameAs(callback1));
+            Assert.That(conn2.UserCertificateValidationCallback, Is.SameAs(callback2));
+            conn2.Open();
         }
 
         [Test, IssueLink("https://github.com/npgsql/npgsql/issues/824")]
         public void ReloadTypes()
         {
-            var connString = new NpgsqlConnectionStringBuilder(ConnectionString)
+            var csb = new NpgsqlConnectionStringBuilder(ConnectionString)
             {
                 ApplicationName = nameof(ReloadTypes)
-            }.ToString();
-            using (var conn = OpenConnection(connString))
-            using (var conn2 = OpenConnection(connString))
-            {
-                Assert.That(conn.ExecuteScalar("SELECT EXISTS (SELECT * FROM pg_type WHERE typname='reload_types_enum')"),
-                    Is.False);
-                conn.ExecuteNonQuery("CREATE TYPE pg_temp.reload_types_enum AS ENUM ('First', 'Second')");
-                Assert.That(() => conn.TypeMapper.MapEnum<ReloadTypesEnum>(), Throws.Exception.TypeOf<ArgumentException>());
-                conn.ReloadTypes();
-                conn.TypeMapper.MapEnum<ReloadTypesEnum>();
-
-                // Make sure conn2 picks up the new type after a pooled close
-                var connId = conn2.ProcessID;
-                conn2.Close();
-                conn2.Open();
-                Assert.That(conn2.ProcessID, Is.EqualTo(connId), "Didn't get the same connector back");
-                conn2.TypeMapper.MapEnum<ReloadTypesEnum>();
-
-                NpgsqlConnection.ClearPool(conn);
-            }
+            };
+            using var conn = OpenConnection(csb.ConnectionString);
```

Comment:
csb.ConnectionString twice is less good than once :)

Reply:
The value of `ConnectionString` is cached, so no, it's still good to use it.

Reply:
It's not about perf (in any case we're in tests), it's just about not repeating code for no reason

Reply:
Then in other places we should store `ConnectionString` in locals too just for the common style, but I got the idea. Will make `OpenConnection` accepting a connection string builder.